lib/protoflow-blocks/src/blocks/flow/concat.rs

```diff
@@ -0,0 +1,188 @@
+// This is free and unencumbered software released into the public domain.
+extern crate std;
+
+use crate::prelude::{Arc, Vec};
+use crate::{FlowBlocks, StdioConfig, StdioError, StdioSystem, SysBlocks, System};
+use protoflow_core::{
+    types::Any, Block, BlockResult, BlockRuntime, InputPort, Message, OutputPort,
+};
+use protoflow_core::{BlockError, PortError};
+use protoflow_derive::Block;
+use simple_mermaid::mermaid;
+
+/// Combines multiple input streams into a single output stream in sequence.
+///
+/// # Block Diagram
+#[doc = mermaid!("../../../doc/flow/concat.mmd")]
+///
+/// # Sequence Diagram
+#[doc = mermaid!("../../../doc/flow/concat.seq.mmd" framed)]
+///
+/// # Examples
+///
+/// ## Using the block in a system
+///
+/// ```rust
+/// # use protoflow_blocks::*;
+/// # fn main() {
+/// System::build(|s| {
+///     let stdin = s.read_stdin();
+///
+///     let replicate = s.replicate();
+///     s.connect(&stdin.output, &replicate.input);
+///
+///     let concat = s.block(Concat::new(s.input(), s.input(), s.output()));
+///     s.connect(&replicate.output_1, &concat.input_1);
+///     s.connect(&replicate.output_2, &concat.input_2);
+///
+///     let stdout_1 = s.write_stdout();
+///     s.connect(&concat.output, &stdout_1.input);
+/// });
+/// # }
+/// ```
+///
+/// ## Running the block via the CLI
+///
+/// ```console
+/// $ protoflow execute Concat
+/// ```
+///
+#[derive(Block, Clone)]
+pub struct Concat<T: Message = Any> {
+    /// The input message stream.
+    #[input]
+    pub input_1: InputPort<T>,
+    #[output]
+    pub input_2: InputPort<T>,
+    #[output]
+    pub output: OutputPort<T>,
+}
+
+impl<T: Message> Concat<T> {
+    pub fn new(input_1: InputPort<T>, input_2: InputPort<T>, output: OutputPort<T>) -> Self {
+        Self {
+            input_1,
+            input_2,
+            output,
+        }
+    }
+}
+impl<T: Message + 'static> Concat<T> {
+    pub fn with_system(system: &System) -> Self {
+        use crate::SystemBuilding;
+        Self::new(system.input(), system.input(), system.output())
+    }
+}
+
+impl<T: Message + Send + 'static> Block for Concat<T> {
+    fn execute(&mut self, runtime: &dyn BlockRuntime) -> BlockResult {
```

Comment:
This doesn't need to spawn threads for receiving from the inputs. The [document](https://jpaulm.github.io/fbp/morrison_2005.htm) that we're basing this on describes that `Concat` receives all from first port and then starts receiving from the second port and how they solve the deadlock possibility with a `InfQueue` block in-between.

I'd suggest such with whatever tracing messages you want added:
```rust
    fn execute(&mut self, runtime: &dyn BlockRuntime) -> BlockResult {
        runtime.wait_for(&self.output)?;
        let mut messages = Vec::new();
        while let Some(message) = self.input_1.recv()? {
            messages.push(message);
        }
        while let Some(message) = self.input_2.recv()? {
            messages.push(message);
        }
        for message in messages {
            self.output.send(&message)?;
        }
        Ok(())
    }
```

Reply:
If we assume that all messages are received from input 1 and then messages are received from input 2 the block becomes less useful. As an example; we've made a call to 2 different llm. since we can not know which will respond first, we won't be able to use this block to concatenate. The drawback here is spawning threads. So let @artob decide.